### PR TITLE
fix: use http/1.1 for post

### LINKED ISSUE #5066: owlbot hitting commit size limit
```
Error: Command failed: git push -f origin owl-bot-copy-java-common-protos
Step #3: error: RPC failed; HTTP 408 curl 18 HTTP/2 stream 7 was reset
```

### PR DESCRIPTION
**Pull request by bcoe**
Attempt the resolution documented [here](https://github.com/orgs/community/discussions/55820) for ` RPC failed; HTTP 408 curl 18 HTTP/2 stream 7 was reset`.

Fixes #5066

### CODE REVIEW

File: packages/owl-bot/cloud-build/scan-googleapis-gen-and-create-pull-requests.yaml [hunk 1]
```diff
@@ -33,6 +33,28 @@ steps:
       - --global
       - user.name
       - 'Owl Bot'
+  # TODO: remove when we find a better resolution.
+  # See: https://github.com/orgs/community/discussions/55820
+  - name: 'gcr.io/cloud-builders/git'
```
Comment (chingor13) [inline packages/owl-bot/cloud-build/scan-googleapis-gen-and-create-pull-requests.yaml:38]: I don't think this method is going to work. GCB only shares the /workspace directory between steps (or possibly other folders if you explicitly share). `git config --global` commands puts config in your `~/.git/config`.

**Review by chingor13 — CHANGES_REQUESTED** [review]
We should put this in the owlbot-cli Dockerfile as it's a global config

File: packages/owl-bot/cloud-build/scan-googleapis-gen-and-create-pull-requests.yaml (hunk 1 above)
Comment (bcoe) [reply packages/owl-bot/cloud-build/scan-googleapis-gen-and-create-pull-requests.yaml:38]: I believe the git push happens in the CLI command not in the docker container:

https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/src/copy-code.ts#L817

**Review by chingor13 — APPROVED** [review]

File: packages/owl-bot/Dockerfile [hunk 2]
```diff
@@ -31,6 +31,11 @@ COPY . ./
 RUN apk update && apk upgrade && \
     apk add --no-cache bash git openssh
 
+# TODO: remove when we find a better resolution.
+# See: https://github.com/orgs/community/discussions/55820
+RUN git config --global http.version HTTP/1.1 \
```
Comment (chingor13) [inline packages/owl-bot/Dockerfile:36]: need an `&&`

**Review by chingor13 — CHANGES_REQUESTED** [review]

File: packages/owl-bot/Dockerfile (hunk 2 above)
Comment (bcoe) [reply packages/owl-bot/Dockerfile]: whoops, done. out of practice.

**Review by chingor13 — APPROVED** [review]

**Pull request merged**